src/kernel/src/ucred/prison.rs

```diff
@@ -0,0 +1,77 @@
+use bitflags::bitflags;
+use std::borrow::Cow;
+
+#[derive(Debug)]
+pub struct Prison {
+    parent: Option<Cow<'static, Self>>,
+    flags: PrisonFlags,
+    allow: PrisonAllow,
+}
+
+impl Prison {
+    pub fn flags(&self) -> PrisonFlags {
+        self.flags
+    }
+
+    pub fn allow(&self) -> PrisonAllow {
+        self.allow
+    }
+
+    pub fn is_child(&self, other: &Self) -> bool {
+        let mut p = other.parent.as_ref();
+
+        while let Some(pr) = p {
+            if self == other {
+                return true;
+            }
+
+            p = pr.parent.as_ref();
+        }
+
+        false
+    }
+}
+
+impl PartialEq for Prison {
+    fn eq(&self, other: &Self) -> bool {
+        std::ptr::eq(self, other)
+    }
+}
```

Comment:
Compare a pointer is not really suitable with Rust. We should have some mechanism to generate a unique ID for each `Prison` and use that to identify instead.